glide-core/src/client/mod.rs

```diff
@@ -474,6 +476,33 @@ impl Client {
         self.inflight_requests_allowed
             .fetch_add(1, Ordering::SeqCst)
     }
+
+    /// Replaces the current password with the provided ones.
+    /// If `re_auth` is true, the new password will be used to re-authenticate with the server.
+    pub async fn replace_connection_password(
+        &mut self,
+        password: String,
+        re_auth: bool,
+    ) -> RedisResult<Value> {
+        if re_auth {
+            let routing = RoutingInfo::MultiNode((
```

Comment:
I vaguely remember in EC (non Serverless) that there are cases that different AUTH token is used per shard.